api/module.js

```diff
@@ -158,9 +178,7 @@ export const COMMONJS_WRAPPER = CommonJSModuleScope
  * @type {string}
  */
 export const MAIN_SOURCE_URL = (
-  globalThis.origin === 'file://' && globalThis.location?.href
-    ? globalThis.location.href
-    : `file://${process.cwd() || ''}`
+  globalThis.location?.href || `file://${process.cwd() || ''}`
```

Comment:
`process.cwd()` can change.. I'm not sure if an import could be triggered after user space code has modified though cwd though.... Maybe a userspace import could modify it, which would be a bad idea, but:
What about storing the starting cwd on app start, and store this on process? Accessible via `process.startDir()` perhaps.

Reply:
we don't implement `process.chdir()` method and `process.cwd()` is always returns `primordials.cwd` which we can use here as well